app/connectors/MessageConnector.scala

```diff
@@ -32,21 +34,36 @@ class MessageConnector @Inject()(http: HttpClient, appConfig: AppConfig) extends
 
   val rootUrl = appConfig.traderAtDestinationUrl + "/transit-movements-trader-at-destination/movements"
 
-  def get(arrivalId: String, messageId: String)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Either[HttpResponse, MovementMessage]] = {
+  def get(arrivalId: String, messageId: String, headers: Headers)(implicit headerCarrier: HeaderCarrier, ec: ExecutionContext): Future[Either[HttpResponse, MovementMessage]] = {
     val url = rootUrl + s"/arrivals/${Utils.urlEncode(arrivalId)}/messages/${Utils.urlEncode(messageId)}"
-    http.GET[HttpResponse](url, queryParams = Seq(), responseHeaders())(CustomHttpReader, implicitly, implicitly).map { response =>
+
+    val newHeaders = headerCarrier
+      .copy(authorization = Some(Authorization(headers.get("Authorization").getOrElse(""))))
+      .withExtraHeaders(responseHeaders(): _*)
+
+    http.GET[HttpResponse](url, queryParams = Seq(), responseHeaders())(CustomHttpReader, hc = newHeaders, ec = ec).map { response =>
       extractIfSuccessful[MovementMessage](response)
     }
   }
 
-  def post(message: String, arrivalId: String)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[HttpResponse] = {
+  def post(message: String, arrivalId: String, headers: Headers)(implicit headerCarrier: HeaderCarrier, ec: ExecutionContext): Future[HttpResponse] = {
```

Comment:
You already have the headercarrier object. You don't need to pass in the headers seperately, you can use `headerCarrier.headers(HeaderNames.AUTHORIZATION)`